### DISCUSSION TITLE
Where to find how the NRMSE is Calculated in GluonTS

### DISCUSSION THREAD

**Discussion by MarleneBs** [awslabs/gluonts#2336] [Q&A] [answered]
Can anyone tell me how the NRMSE is calculated in GLuonTS, in particular i am looking for the formula.
I have through general knowledge the knowledge about how the RMSE but I don't know how gluonts normalizes that now, but I need to know the calculation rule for my work.

**Comment by jgasthaus** [comment, marked as answer]
If you have $N$ time series, and a prediction length of $H$, denote by $x_{i, h}$ the true value of time series $i$ at prediction horizon $h$, and the corresponding mean forecast by $\hat{x}_{i, h}$. Then the RMSE is computed as 
$$\text{RMSE} = \sqrt{\frac{1}{N}\sum\_{i=1}^N \frac{1}{H}\sum\_{h=1}^{H} (x\_{i, h} - \hat{x}\_{i, h})^2}.$$
The NRMSE is RMSE normalized by the mean of the absolute true values, i.e. 
$$\text{NRMSE} = \frac{\text{RMSE}}{\frac{1}{NH}\sum\_i \sum\_h |x\_{i,h}|}$$

The corresponding code is split across multiple places: 
* The inner mean squared error for each time series is computed by the `mse` function: https://github.com/awslabs/gluonts/blob/015ff06ae78ef78d7f5d73ccceef7ceb9d4db095/src/gluonts/evaluation/metrics.py#L55
* This per-time series MSE is then averaged across time series by using `mean` as the aggregation function here: https://github.com/awslabs/gluonts/blob/015ff06ae78ef78d7f5d73ccceef7ceb9d4db095/src/gluonts/evaluation/_base.py#L432
* The square-root to obtain RMSE is applied here: https://github.com/awslabs/gluonts/blob/015ff06ae78ef78d7f5d73ccceef7ceb9d4db095/src/gluonts/evaluation/_base.py#L464
* NRMSE is obtained by dividing by the mean of absolute target values here: https://github.com/awslabs/gluonts/blob/015ff06ae78ef78d7f5d73ccceef7ceb9d4db095/src/gluonts/evaluation/_base.py#L465

**Reply by MarleneBs** [reply]
Thanks, great I didn't find that, that's exactly the answer I needed! Thank you very much.
Does the abs_target_mean mean the mean value of the comparison values, that is the true values or does it refer to the predicted values? 
One more question, according to this, is it also possible that the NRMSE is greater than 1?
